src/runtime_src/core/pcie/linux/shim.cpp

```diff
@@ -1337,7 +1337,11 @@ int shim::xclLoadXclBin(const xclBin *buffer)
         xrt_logmsg(XRT_ERROR, "CU was deadlocked? Hardware is not stable");
         xrt_logmsg(XRT_ERROR, "Please reset device with 'xbutil reset'");
       }
-      xrt_logmsg(XRT_ERROR, "See dmesg log for details. err=%d", ret);
+      else if (ret == -EINVAL) {
+        xrt_logmsg(XRT_ERROR, "Trying to load/program invalid xclbin");
+      }
+      xrt_core::message::send(xrt_core::message::severity_level::error, "XRT",
+         std::string("See dmesg log for details. err = " + std::to_string(ret)));
```

Comment:
Please fix xrt_logmsg, in particular fix the bug in xrt_core::message::send in message.h.  It smells like an off-by-one some where.  Working around the bug in one place is not a good solution given that many places use xrt_logmsg.  Get to the root of the problem, please.

Turns out the bug is in max number of bytes passed to snprintf; it is indeed off by one.  
```
    std::vector<char> buf(sz+1);
    snprintf(buf.data(), sz, format, args ...);
    send(l, tag, buf.data());
```
Since snprintf adds the null terminator to the buffer, the buffer is allocated with room for that, but the call to snprintf doesn't reflect the actual size of the buffer, and hence is chopped off one byte.  So, should be:
```
    std::vector<char> buf(sz+1);
    snprintf(buf.data(), sz+1, format, args ...);
    send(l, tag, buf.data());
```

Reply:
Thanks @stsoe , I completely agree with you. As a starting point, I have just replaced in once shim.cpp because I don't want to disturb already working code/setup. 

I have added fix in message.h file now.